contrib/dyn_templates/src/lib.rs

```diff
@@ -439,3 +459,70 @@ impl Sentinel for Template {
         false
     }
 }
+
+/// A macro to easily create a template rendering context.
+///
+/// # Example
+///
+/// This code:
+///
+/// ```rust
+/// # #[macro_use] extern crate rocket;
+/// # use rocket_dyn_templates::{Template, context};
+///
+/// #[get("/<param>")]
+/// fn render_index(param: u64) -> Template {
+///     Template::render("index", context!{
+///         foo: "Hello world",
+///         bar: param,
+///     })
+/// }
+/// ```
+///
+/// is equivalent to the following, but without the need to manually define an
+/// `IndexContext` struct:
+///
+/// ```rust
+/// # #[macro_use] extern crate rocket;
+/// # use rocket_dyn_templates::Template;
+/// # use serde::Serialize;
+/// #[derive(Serialize)]
+/// struct IndexContext<'a> {
+///     foo: &'a str,
+///     bar: u64,
+/// }
+///
+/// #[get("/<param>")]
+/// fn render_index(param: u64) -> Template {
+///     Template::render("index", IndexContext {
+///         foo: "Hello world",
+///         bar: param,
+///     })
+/// }
+/// ```
+#[macro_export]
+macro_rules! context {
+    ($($key:ident $(: $value:expr)?),*$(,)?) => {{
+        use $crate::serde_internal::ser::{Serialize, Serializer, SerializeMap};
+
+        #[allow(non_camel_case_types)]
+        struct AnonymousInnerContextStruct<$($key: Serialize),*> {
```

Comment:
We should have some UI tests for when values aren't Serialize. I think having this bound here might result in suboptimal error messages. It should almost be unnecessary.

Reply:
Removing that trait bound allows code to compile if values *aren't* `Serialize`, as long as the `Serialize` impl of the returned value isn't used. i.e. this compiles fine:
```rust
enum NonSerialize { A, B, C }

let ctx = context! {
    foo: NonSerialize::A,
};
```
but not this:
```rust
enum NonSerialize { A, B, C }

let ctx = context! {
    foo: NonSerialize::A,
};
Template::render("index", ctx);
```
As of now, the error messages are actually quite good:
```
error[E0277]: the trait bound `NonSerialize: Serialize` is not satisfied
  --> src/main.rs:18:11
   |
18 |   let ctx = context! {
   |  ___________^
19 | |     foo: NonSerialize::A,
20 | | };
   | | ^
   | | |
   | |_the trait `Serialize` is not implemented for `NonSerialize`
   |   required by `ContextMacroCtxObject`
```

I'll add some ui tests. Where/how should I add them?